### PR TITLE
Refactor production blocks registration 5

### PR DESCRIPTION
**Pull request by monsieur-z**
- See https://github.com/Automattic/jetpack/issues/32602
- See https://github.com/Automattic/jetpack/issues/33174

## Proposed changes:

Registers the following production blocks by specifying the path to their block.json file:  _Related Posts_, _Repeat Visitor_, _Slideshow_, _Story_.

This was made possible by https://github.com/Automattic/jetpack/pull/32697. The goal is to avoid the duplication of block metadata and use block.json as the source of truth.

In more details, for each block (when needed):
  - Write reading direction-agnostic CSS
  - Remove the `editorScript` property from `block.json` (automatically added during the build)
  - Convert icon from dashicon to SVG in the `icon` property of `block.json`
  - Remove `icon.js`
  - Import icon helpers from the `shared-extension-utils` package
  - Register the block client and server sides with `block.json`
  - Move data from `index.js` to `block.json` and `editor.js` and delete the file
  - Move data from `attributes.js` to `block.json` and delete the file
  - Fix JS tests

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pedMtX-RS-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Spin up a test site
- Create a new post and open the block inserter
- Test that the aforementioned blocks behave as in `trunk`. This PR shouldn't change anything from a user's standpoint.

### CODE REVIEW

File: projects/plugins/jetpack/tools/check-block-assets.php [hunk 1]
```diff
@@ -43,6 +43,7 @@
 	'wp-html-entities',
 	'wp-i18n',
 	'wp-is-shallow-equal',
+	'wp-plugins',
```
Comment (jeherve) [inline projects/plugins/jetpack/tools/check-block-assets.php]: Did you add this in for the Story block? If we need to add this in here, maybe we could allow it just for the Story block, with a change to `$allowed`?

**Review by jeherve — COMMENTED** [review]
This tests well. I only have a small question.

**Review by jeherve — COMMENTED** [review]
I'll let someone else review that one (@andrii-lysenko maybe), because I'm running into an issue with the Pay with PayPal block, and I'm not sure if it's something to do with my setup. I am experiencing the same problem when on `trunk`, and only on self-hosted (a new JN site and an existing WoA site). It works well on wpcom simple.

The block seems to work well in the editor, but I do not see it on the frontend. And when refreshing the block editor, the data in the block disappears.

**Comment by monsieur-z** [comment]
UPDATE: I moved all payment-related changes to a separate PR. This PR now effectively changes the following blocks: Related Posts, Repeat Visitor, Slideshow, Story.

**Review by jeherve — APPROVED** [review]

**Pull request merged**